### PR TITLE
PADV-419 - SCORM xblock migration.

### PR DESCRIPTION
**Pull request by alexjmpb**
## Ticket

https://agile-jira.pearson.com/browse/PADV-419

## Description

This PR aims to add the necessary changes to make the scorm xblock compatible with edx-platform olive version.

## Type of Change

- [x] Make compatibility changes
- [x] Create MD file to justify the reasons behind the changes

## Testing:

- Clone repo on devstack src folder
- Install the xblock in the LMS and CMS containers
- Download a scorm package (you can do so here https://scorm.com/scorm-explained/technical-scorm/golf-examples/?utm_source=google&utm_medium=natural_search)
- Then create a new course with the 'scormxblock' in the advanced modules enabled and create a new unit uploading the package
- Go to the course and verify that the xblock works correctly

## Screenshots:

image
image

## Reviewers

- [ ] @Squirrel18 
- [x] @anfbermudezme

### CODE REVIEW

File: scormxblock/scormxblock.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def updoad_all_content(temp_directory, fs):
     """
     This standalone function handles the bulk upload of unzipped content.
     """
-    if not settings.DJFS.get('type', 'osfs') == "s3fs":
+    if settings.DJFS.get('type', 'osfs') != "s3fs":
```
Comment (anfbermudezme) [inline scormxblock/scormxblock.py:141]: I think `not` it's more explicit.

File: scormxblock/scormxblock.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def updoad_all_content(temp_directory, fs):
 
     if len(all_content) < FILES_THRESHOLD_FOR_ASYNC:
         # We estimate no problem here, just upload the files
-        s3_upload(all_content, temp_directory, dest_dir)
+        s3_upload.delay(all_content, temp_directory, dest_dir)
```
Comment (anfbermudezme) [inline scormxblock/scormxblock.py:155]: Why this change is necessary?

File: scormxblock/scormxblock.py [hunk 3]
```diff
@@ -251,6 +253,7 @@ def resource_string(self, path):
 
     @XBlock.supports('multi_device') # Mark as mobile-friendly
     def student_view(self, context=None):
+        # create docstring for this method
```
Comment (anfbermudezme) [inline scormxblock/scormxblock.py]: I guess you missed this docstring.

File: scormxblock/scormxblock.py [hunk 4]
```diff
@@ -308,7 +324,7 @@ def studio_submit(self, request, suffix=''):
             # Destroy temp directory after all files are copied.
             temp_directory.close()
 
-        return Response({'result': 'success'}, content_type='application/json')
+        return Response({'result': 'success'}, content_type='application/json', charset='utf8')
```
Comment (anfbermudezme) [inline scormxblock/scormxblock.py:326]: Why is necessary specify the character encoding?

File: scormxblock/scormxblock.py (hunk 1 above)
Comment (alexjmpb) [reply scormxblock/scormxblock.py:141]: I changed it because I think is more immediate to notice what the line does with de !=. It happened to me that I didn't understand the line because i was looking only at ==. But it can be me, what do you think?

File: scormxblock/scormxblock.py (hunk 2 above)
Comment (alexjmpb) [reply scormxblock/scormxblock.py:155]: It was not running in a external worker. I think now the task has to be called with delay

File: scormxblock/scormxblock.py (hunk 4 above)
Comment (alexjmpb) [reply scormxblock/scormxblock.py:326]: Now the Reponse class requires to specify the charset otherwise it throws an error

**Review by anfbermudezme — APPROVED** [review]

**Review by sergivalero20 — APPROVED** [review]

**Review by Squirrel18 — APPROVED** [review]
